input.c

```diff
@@ -90,12 +90,38 @@ modify_words(const char *val, int modifier, const char *find, const char *repl)
 					word = sep + 1;
 				}
 			}
-			if (repl) {
-				size_t lenw = strlen(word);
-				size_t lenf = strlen(find);
-				if (lenw >= lenf && strcmp(word + lenw - lenf, find) == 0) {
-					word[lenw - lenf] = '\0';
-					word = newword = xconcat3(word, repl, "");
+			size_t lenw = strlen(word);
+			size_t lenf = strlen(find);
+			if (repl && lenw >= lenf) {
+#if ENABLE_FEATURE_MAKE_EXTENSIONS
+				char *perc_find = strchr(find, '%');
+				if (perc_find) {
+					size_t pref_find = perc_find - find;
+					size_t suff_find = strlen(perc_find + 1);
+					if (strncmp(word, find, pref_find) == 0 &&
+						strcmp(word + lenw - suff_find, perc_find + 1) == 0) {
+						char *perc_repl = strchr(repl, '%');
+						if (!perc_repl) {
+							word = newword = xstrdup(repl);
+						} else {
+							size_t pref_repl = perc_repl - repl;
+							size_t suff_repl = strlen(perc_repl + 1);
+							size_t word_base = lenw - (suff_find + pref_find);
+							size_t len = pref_repl + word_base + suff_repl + 1;
+
+							newword = xmalloc(len);
+							snprintf(newword, len, "%.*s%.*s%s", pref_repl, repl,
```

Comment:
the arguments to the `*`s here should be cast to `int`, because that's the expected argument rather than `size_t`

Reply:
Hate this thing...Fixed thanks